### PR TITLE
fix: don't allow session edit after session is held

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/5051
related https://github.com/fossasia/open-event-frontend/pull/5959

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/sessions.py [hunk 1]
```diff
@@ -267,6 +269,12 @@ def before_update_object(self, session, data, view_kwargs):
                 "Locked sessions cannot be edited",
             )
 
+        if(session.ends_at and session.ends_at < datetime.now()):
```
Comment (iamareebjamal) [inline app/api/sessions.py:272]: They can be edited, but not withdrawn.

**Comment by iamareebjamal** [comment]
Organizer should be able to edit after session is held

**Comment by maze-runnar** [comment]
but here I am checking only for state change is allowed or not.

**Comment by iamareebjamal** [comment]
Yes, after this change, organizer cannot edit the state of the session

**Comment by maze-runnar** [comment]
why test is failing ? 🤦🏻

**Review by iamareebjamal — APPROVED** [review]

**Comment by iamareebjamal** [comment]
Tests failing

**Comment by maze-runnar** [comment]
> Tests failing

I am not getting why test is failing?

**Comment by iamareebjamal** [comment]
https://app.circleci.com/pipelines/github/fossasia/open-event-server/1506/workflows/c17dd8e6-628e-40f6-be97-6147b6c0d715/jobs/5911

**Comment by iamareebjamal** [comment]
Test the feature locally, it won't work as well

**Comment by iamareebjamal** [comment]
No, read the error message. Try locally

File: app/api/sessions.py [hunk 2]
```diff
@@ -269,7 +270,10 @@ def before_update_object(self, session, data, view_kwargs):
 
         new_state = data.get('state')
 
-        if new_state and new_state != session.state:
+        if new_state and new_state != session.state and (
+            is_organizer or (
+            session.ends_at and 
+            session.ends_at.date() > datetime.now().date())):
```
Comment (iamareebjamal) [inline app/api/sessions.py:276]: Still wrong. Read the error carefully

Comment (maze-runnar) [reply app/api/sessions.py:276]: test is fixed. And its working correctly too

Comment (iamareebjamal) [reply app/api/sessions.py:276]: session.ends_at > datetime.utcnow()

**Pull request merged**